### PR TITLE
Updates to Windows provisioning document

### PR DESCRIPTION
**Pull request by yying**

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

File: docs/wiki/development/windows-provisioning.md [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
-The PowerShell script `provision.ps1` is used to prepare a clean Windows 10 64 bit machine into one that is ready of **osquery** development on Windows. However, the script does _not_ automate the generation of the Visual Studio 2015 solution nor performs the build process.
+The bulk of the development environment provisioning logic is in the `tools\provision.ps1` PowerShell script. It is **not** recommended to directly invoke this script. Instead, run the `tools\make-win64-dev-env.bat` batch script. This provisioning script is used to prepare a clean Windows 10 64 bit operating system into one that is prepared for **osquery** development by downloading and installing the proper tools and dependencies.
+
+Generating the Visual Studio 2015 solution and building the **osquery** binaries is done via the `tools\make-win64-binaries.bat` batch script. 
 
 ## Initial Assumptions
 
  * `git` for Windows should be already installed in order to `git clone` the **osquery** repository containing the provisioning script.
- * The machine is running the Windows 10 64 bit operating system with PowerShell
+ * The machine is running the Windows 10 64 bit operating system with PowerShell installed
```
Comment (PoppySeedPlehzr) [inline docs/wiki/development/windows-provisioning.md]: This is somewhat implicitly assumed, but Powershell 3.0 is expected.  Would you mind adding a `>= 3.0` or the like for folks who decide to get adventurous with where they provision?

**Comment by facebook-github-bot-9** [comment]
@yying updated the pull request - [view changes](https://github.com/facebook/osquery/pull/2471/files/ed4bad52f732282ddad1fdb25f0c071d1c258216..e85ca0e938d18d266a61061b83e6406567fd3461)

File: docs/wiki/development/windows-provisioning.md [hunk 2]
```diff
@@ -1,15 +1,26 @@
...
+The bulk of the development environment provisioning logic is in the `tools\provision.ps1` PowerShell script. It is **not** recommended to directly invoke this script. Instead, run the `tools\make-win64-dev-env.bat` batch script. This provisioning script is used to prepare a clean Windows 10 64 bit operating system into one that is prepared for **osquery** development by downloading and installing the proper tools and dependencies.
+
+Generating the Visual Studio 2015 solution and building the **osquery** binaries is done via the `tools\make-win64-binaries.bat` batch script. 
 
 ## Initial Assumptions
 
  * `git` for Windows should be already installed in order to `git clone` the **osquery** repository containing the provisioning script.
- * The machine is running the Windows 10 64 bit operating system with PowerShell
+ * The machine is running the Windows 10 64 bit operating system with PowerShell 3.0 and later installed
```
Comment (theopolis) [inline docs/wiki/development/windows-provisioning.md]: nit, word choice: The machine is running Windows 10 64-bit with PowerShell 3.0 or later

File: docs/wiki/development/windows-provisioning.md [hunk 3]
```diff
@@ -20,25 +31,20 @@ The PowerShell script `provision.ps1` is used to prepare a clean Windows 10 64 b
  * Generate the Visual Studio 2015 solution files: `cmake ..\.. -G "Visual Studio 14 2015 Win64"`
  * There should be a `OSQUERY.sln` in the build folder. Open this with Visual Studio 2015 that is already installed via the provisioning script.
 
-## Building `osqueryd.exe` and `osqueryi.exe`
+### Building `osqueryd.exe` and `osqueryi.exe`
```
Comment (theopolis) [inline docs/wiki/development/windows-provisioning.md]: In the above two sections, this workflow is repeated a few times. This is a good opportunity to clean up it.

File: docs/wiki/development/windows-provisioning.md [hunk 4]
```diff
@@ -1,44 +1,46 @@
...
+Generating the Visual Studio 2015 solution and building the **osquery** binaries is done via the `tools\make-win64-binaries.bat` batch script. 
 
 ## Initial Assumptions
 
  * `git` for Windows should be already installed in order to `git clone` the **osquery** repository containing the provisioning script.
- * The machine is running the Windows 10 64 bit operating system with PowerShell
- * No previous instance of Visual Studio 2015 is already installed.
- * The user is running the script as an **Administrator**
+ * The machine is running the Windows 10 64 bit operating system with PowerShell 3.0 or later installed
+ * No previous instance of Visual Studio 2015 is installed.
```
Comment (theopolis) [inline docs/wiki/development/windows-provisioning.md]: This page has an inconsistency of `.` usage. If you don't mind, can we have all the list elements either use a `.` or have no punctuation?

File: docs/wiki/development/windows-provisioning.md [hunk 5]
```diff
@@ -1,44 +1,46 @@
...
    
 ## Chocolatey Packages Installed (from official sources)
 
  * chocolatey (if applicable)
  * 7zip.commandline
- * cmake.portable 3.5.0
+ * cmake.portable 3.6.1
  * python2 2.7.11
  * visualstudio2015community (with a custom deployment XML ensuring C/C++ toolchain is installed)
  * thrift 0.9.3 (as a dependency from one of our private packages)
```
Comment (theopolis) [inline docs/wiki/development/windows-provisioning.md]: If `thrift` is from a private package, does that mean in should be move to the next section? How about the `visualstudio2015community`?

Comment (yying) [inline docs/wiki/development/windows-provisioning.md]: thrift is actually from chocolatey repo, visualstudio2015community should remain

I think I'll remove the content in the parens to make things simple

**Comment by facebook-github-bot-2** [comment]
@yying updated the pull request - [view changes](https://github.com/facebook/osquery/pull/2471/files/78d171f0aefb38abe789e4e1fdd2d738e6cc2c31..d568eadbc43489d36b24a598ddf415e91731fd9e)

**Comment by theopolis** [comment]
👍

**Pull request merged**